src/lib/CheerpJ.svelte

```diff
@@ -1,40 +1,37 @@
 <script lang="ts">
-	import { createEventDispatcher } from 'svelte';
 	import { browser } from '$app/environment';
-	import { tryPlausible } from './plausible';
-	import { files, loading, type File } from './repl/state';
+	import { onDestroy, onMount } from 'svelte';
+	import { autoRun, compileLog, files, loading, isSaved, runCode, type File } from './repl/state';
+	import { debounceFunction } from './utilities';
 
-	const dispatch = createEventDispatcher<{ ready: undefined }>();
-
-	export let display: HTMLElement;
-	export let consoleEl: HTMLPreElement;
-
-	async function onLoad() {
+	let display: HTMLElement | null;
+	let consoleEl: HTMLElement | null;
+	let cjOutputObserver: MutationObserver;
 
+	async function startCheerpj() {
 		await cheerpjInit({
 			status: 'none',
 			javaProperties: ['java.library.path=/app/cheerpj-natives/natives']
 		});
 		cheerpjCreateDisplay(-1, -1, display);
-		dispatch('ready');
-	}
-
-	if (browser) {	// so it doesn't run server-side
-		onLoad();
+		// remove useless loading screen
+		// console.log(`display: ${display.innerHTML}`);
+		document.getElementById("cheerpjDisplay")?.classList.remove("cheerpjLoading");
```

Comment:
This should not be conditional `?`. The element is guaranteed to exist and was just created

Also remove the console log